common/lib/xmodule/xmodule/html_module.py

```diff
@@ -115,10 +115,21 @@ def student_view_data(self, context=None):  # pylint: disable=unused-argument
             }
 
     def get_html(self):
-        """ Returns html required for rendering the block. """
-        if self.data is not None and getattr(self.system, 'anonymous_student_id', None) is not None:
-            return self.data.replace("%%USER_ID%%", self.system.anonymous_student_id)
-        return self.data
+        """
+        Returns HTML required for rendering the block.
+        """
+        data = self.data
+        user = None
+        
+        if data is not None and getattr(self.system, 'anonymous_student_id', None) is not None:
+            data = self.data.replace("%%USER_ID%%", self.system.anonymous_student_id)
```

Comment:
Suggestion: Maybe we could do this instead `is not None`
`if data and getattr(self.system, 'anonymous_student_id', None):`